File: BH Tic Tac Toe IntelliJ Project/src/org/bh/game/ttt/game/Player.java
package org.bh.game.ttt.game;

/**
 * Player, made for BH Tic Tac Toe NetBeans Project, is copyright Blue Husky Programming ©2014 GPLv3 <hr/>
 * 
 * @author Kyli of Blue Husky Programming
 * @version 1.0.0
 *		- 2014-09-23 (1.0.0) - Kyli created Player
 * @since 2014-09-23
 */
public class Player
{
	private String name;
	private char symbol;

	public Player(String name, char symbol)
	{
		this.name = name;
		this.symbol = symbol;
	}

	public String getName()
	{
		return name;
	}

	public void setName(String name)
	{
		this.name = name;
	}

	public char getSymbol()
	{
		return symbol;
	}

	public void setSymbol(char symbol)
	{
		this.symbol = symbol;
	}
	
	/**
	 * Asks the player for their next move. If the player is going to take a while, {@code null} is returned, indicating that
	 * the caller should set up a listener for the user's action and wait for that in order to assess the move.
	 * 
	 * @return The next move, or {@code null} if that will take a while.
	 */
	public Move nextMove()
	{
		return null;
	}
}

File: BH Tic Tac Toe IntelliJ Project/src/org/bh/game/ttt/game/Player.kt
package org.bh.game.ttt.game

/**
 * Copyright BHStudios ©2016 BH-1-PS. Made for BH Tic Tac Toe IntelliJ Project.
 *
 * @author Ben C. Leggiero
 * @since 2016-10-10
 */
abstract class Player(var name: String, var symbol: Char) {

    /**
     * Asks the player for their next move. This may be delayed indefinitely.
     */
    abstract fun nextMove(callback: MoveCallback)
}

class __TEST__Player(name: String, symbol: Char) : Player(name, symbol) {
    override fun nextMove(callback: (Move) -> Unit) {
        throw UnsupportedOperationException("not implemented") //To change body of created functions use File | Settings | File Templates.
    }
}

typealias MoveCallback = (move: Move) -> Unit
